content/concepts/did-ddo.md

```diff
@@ -1,173 +1,513 @@
+| **`providerEndpoint`** | `string`                      | **✓**                           | Provider endpoint URI (URI + path)                                                                                                         |
+| **`timeout`**          | `number`                      | **✓**                           | Describing how long the service can be used after consumption is initiated. A timeout of 0 represents no time limit. Expressed in seconds. |
+| **`files`**            | `string`, see [Files](#files) | **✓**                           | Encrypted file URLs.                                                                                                                       |
+| **`privacy`**          | [Privacy](#compute-privacy)   | **✓** (for compute assets only) | If service is of `type` `compute`, holds information about the compute-related privacy settings.                                           |
 
-A part of the `attributes.main` sub-section, other optional sub-sections like `attributes.extra` can be added. These depend on the service type.
+#### Compute Privacy
 
-Each service has a `timeout` (in seconds) section describing how long the sevice can be used after consumption is initiated. A timeout of 0 represents no time limit.
+An asset with a service of `type` `compute` has the following additional attributes under the `privacy` object. This object is required if the asset is of `type` `compute`, but can be omitted for `type` of `access`.
 
-The `cost` attribute is obsolete, as of Ocean V3. As of V3, to consume an asset, one sends exactly 1.0 datatokens of the asset, so a `cost` is not needed.
+| Attribute                                  | Type                                  | Required | Description                                                                                                                                                                                                               |
+| ------------------------------------------ | ------------------------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **`allowRawAlgorithm`**                    | `boolean`                             | **✓**    | If `true`, any passed raw text will be allowed to run. Useful for an algorithm drag & drop use case, but increases risk of data escape through malicious user input. Should be `false` by default in all implementations. |
+| **`allowNetworkAccess`**                   | `boolean`                             | **✓**    | If `true`, the algorithm job will have network access (still WIP)                                                                                                                                                         |
```

Comment:
where exactly is this "still WIP"? In the spec? In implementation? Where?

Reply:
`in implementation` , we should not write that there as it is not relevant to the structure.  we can wright that in the components.